libs/components/tabs/src/lib/modules/tabs/tabset-nav-button.component.ts

```diff
@@ -135,29 +131,19 @@ export class SkyTabsetNavButtonComponent implements OnDestroy {
   #updateButtonVisibility(): void {
     const isLastStep = this.#activeIndexNumber === this.#tabCount - 1;
 
-    if (this.#_buttonType === buttonTypeFinish) {
+    if (this.buttonType === buttonTypeFinish) {
       this.isVisible = isLastStep;
-      return;
-    }
-
-    // Hide the next button on the last step only if a finish button exists.
-    if (
-      this.#_buttonType === buttonTypeNext &&
-      isLastStep &&
-      this.#_tabset.hasFinishButton
-    ) {
-      this.isVisible = false;
-      return;
+    } else if (this.buttonType === buttonTypeNext) {
+      this.isVisible = !isLastStep;
+    } else {
+      this.isVisible = true;
     }
-
-    this.isVisible = true;
   }
 
   #updateButtonProperties(): void {
     if (
-      (this.#_buttonType === buttonTypeNext ||
-        this.#_buttonType === buttonTypeFinish) &&
-      this.#_tabset?.hasFinishButton
+      this.#_buttonType === buttonTypeNext ||
```

Comment:
This code's referencing `#_` properties outside the corresponding get/set accessors.